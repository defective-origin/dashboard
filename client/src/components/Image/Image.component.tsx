import React from 'react'

// ---| core |---
// ---| pages |---
// ---| screens |---
// ---| components |---

// ---| common |---
import { cn } from 'common/tools'

// ---| self |---
import './Image.module.scss'

export const IMAGE_MAP = {
  400: 'https://png.pngtree.com/png-vector/20230416/ourmid/pngtree-unicorn-full-body-beautiful-pattern-png-image_6704420.png',
  500: 'https://d1k5j68ob7clqb.cloudfront.net/processed/with_watermark/6d3H7wV3NUrpA7.png',
  welcome: 'https://www.dndbeyond.com/avatars/thumbnails/30836/227/1000/1000/638063931763028274.png',
}

export type ImageTypes = keyof typeof IMAGE_MAP

export type ImageProps = React.DetailedHTMLProps<React.ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement> & {
  className?: string
  children?: React.ReactNode
  type?: ImageTypes
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <Image />
 */
export function Image(props: ImageProps): JSX.Element {
  const { type, src, children, className, ...otherProps } = props
  const _className = cn('image', className)
  const imgSrc = src ?? IMAGE_MAP[type as ImageTypes]

  return <img className={_className} src={imgSrc} {...otherProps}>{children}</img>
}

Image.displayName = 'Image'

export default Image
