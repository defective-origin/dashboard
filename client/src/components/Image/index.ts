import { Image, ImageProps as AntdImageProps } from 'antd'

export type ImageProps = AntdImageProps
// FIXME: add preview NO IMAGE
export default Image
