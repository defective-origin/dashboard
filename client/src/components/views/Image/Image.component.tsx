import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Skeleton from 'components/views/Skeleton'

// ---| self |---
import './Image.module.scss'

// TODO: preload all images? https://stackoverflow.com/questions/3646036/preloading-images-with-javascript
export const IMAGE_MAP = {
  400: 'https://png.pngtree.com/png-vector/20230416/ourmid/pngtree-unicorn-full-body-beautiful-pattern-png-image_6704420.png',
  500: 'https://d1k5j68ob7clqb.cloudfront.net/processed/with_watermark/6d3H7wV3NUrpA7.png',
  empty: 'https://www.dndbeyond.com/avatars/thumbnails/30836/227/1000/1000/638063931763028274.png',
  error: 'https://png.pngtree.com/png-vector/20231113/ourmid/pngtree-error-rubber-stamp-mistake-png-image_10428685.png',
  logo: 'https://images.crunchbase.com/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco,dpr_1/v1426633865/die6a2gp9buarvojwxk9.png',
}

export type ImageVariant = keyof typeof IMAGE_MAP

export type ImageProps = {
  className?: string
  children?: React.ReactNode
  v?: ImageVariant
  loading?: boolean
  src?: string
  width?: string | number
  height?: string | number
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <Image />
 */
export function Image(props: ImageProps) {
  const { width, height, v, src, loading, children, className, ...otherProps } = props
  const _className = cn('image', className)
  const imgSrc = src ?? IMAGE_MAP[v as ImageVariant]

  const item = (
    <img
      className={_className}
      src={imgSrc}
      style={{ width, height }}
      {...otherProps}
    >
      {children}
    </img>
  )

  if (loading) {
    //TODO: [kseniya_boldak] fix classname
    return <Skeleton className={_className} v='rounded' content={item} />
  }

  return item
}

Image.displayName = 'Image'

export default Image
