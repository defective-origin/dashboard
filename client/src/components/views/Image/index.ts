/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './Image.component'

/* INJECT_EXPORT_PLACE */
export * from './Image.component'
