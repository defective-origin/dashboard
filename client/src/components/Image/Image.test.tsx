// ---| tests |---
import { render } from 'tests'

// ---| self |---
import Image from './Image.component'

describe('[Image] component', () => {
  it('should render component', () => {
    const container = render(<Image />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
