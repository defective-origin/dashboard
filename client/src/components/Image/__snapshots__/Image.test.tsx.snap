// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Image] component > should render component 1`] = `
<DocumentFragment>
  <img
    class="image"
  />
</DocumentFragment>
`;
